solver/src/driver.rs

```diff
@@ -32,16 +34,25 @@ impl Driver {
                 None => return Ok(()),
                 Some(settlement) => settlement,
             };
-        // TODO: encode settlement, check if we need to approve spending to uniswap
+        // TODO: check if we need to approve spending to uniswap
         // TODO: use retry transaction sending crate for updating gas prices
-        self.contract
-            .settle_call(&settlement)
-            .await
-            .context("settle call failed")?;
-        self.contract
-            .settle_send(&settlement)
-            .await
-            .context("settle send failed")?;
+        let settle = || {
+            self.settlement.settle(
```

Comment:
nit: a lot of `settle` here. Maybe we can call it `self.settlement_contract` (eventually there could maybe also be another abstraction between the actual contract and the driver that e.g. handles the retry/gas pricing).